File: APContentAPI/python/apiclient.py
__author__ = 'dalegaspi'

import requests

def main():

    # sign up for developer key at http://developer.ap.org/
    apikey = '** your API key **' 
    
    keyword_search = 'star wars'

    url = 'http://api.ap.org/v2/search/photo'

    payload = requests.get('http://api.ap.org/v2/search/photo',
                           headers={'accept': 'application/json'},
                           params={'apikey': apikey, 'q': keyword_search, 'count': '1'})

    # requests does not automatically handle BOM for UTF-8, hence specifying the encoding
    # with utf-8-sig
    payload.encoding = 'utf-8-sig'
    payload = payload.json()

    suggested_term = payload['suggestedTerm']['title']

    print 'suggested term: "{0}"'.format(suggested_term)
    print 'title of the most recent image about "{0}" is "{1}"'.format(keyword_search, payload['entries'][0]['title'])

    img_url = next(u['href'] for u in payload['entries'][0]['contentLinks'] if u['rel'] == 'thumbnail')

    print 'downloading thumbnail link {0}'.format(img_url)

    img_payload = requests.get(img_url, params={'apikey': apikey})
    if img_payload.status_code == 200:
        f = open('/tmp/thumbnail.jpg', 'wb')
        f.write(img_payload.content)
        f.close()

if __name__ == "__main__":
    main()

File: APContentAPI/dotnet/apiclient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;



namespace AP.APISamples
{
    class APIClient
    {
        static void Main(string[] args)
        {
            // sign up for developer key at http://developer.ap.org/
            var apikey = "** your api key **";

            var keyword_search = "star wars";

            var payload = XElement.Load(string.Format(@"http://api.ap.org/v2/search?count=1&apikey={0}&q={1}", 
                apikey, System.Web.HttpUtility.UrlEncode(keyword_search)));

            XNamespace ns = "http://www.w3.org/2005/Atom";

            var suggested_term = (from s in payload.Elements(ns + "link")
                                 where s.Attribute("rel").Value == "suggestedTerm"
                                 select s.Attribute("title")).FirstOrDefault().Value;

            System.Console.WriteLine("suggested term: \"{0}\"", suggested_term);

            var first_entry = payload.Element(ns + "entry");

            System.Console.WriteLine("title of the most recent image about \"{0}\" is \"{1}\"", keyword_search, first_entry.Element(ns + "title").Value);

            var img_url = (from u in first_entry.Elements(ns + "link")
                                 where u.Attribute("rel").Value == "thumbnail"
                                 select u.Attribute("href")).FirstOrDefault().Value;

            System.Console.WriteLine("downloading thumbnail link {0}", img_url);

            (new System.Net.WebClient()).DownloadFile(img_url + string.Format("&apikey={0}", apikey), @"c:\temp\thumbnail.jpg");
        }
    }
}
